Plugins/Public/balancemagic/Main.cpp

```diff
@@ -455,23 +487,151 @@ void __stdcall HkCb_AddDmgEntry(DamageList *dmg, unsigned short p1, float damage
+					aClientData[iClientID].mapBurstFills[*slot].timestamp = GetTimeInMS() + iter->second.timestamp;
+					ConPrint(L"BURSTFIRE: Client %u fired a round of 0x%08X from slot %u.\n", iClientID, iter->second.ammo, *slot);
+				}
+				//PrintUserCmdText(iClientID, L"FireWeapon: 0x%08X", cgun->GunArch()->iArchID);
+			}
+		}
+	}
+}
+
 void __stdcall SPScanCargo(unsigned int const &iInstigatorShip, unsigned int const &iTargetShip, unsigned int iClientID)
 {
 	returncode = DEFAULT_RETURNCODE;
 }
 
+void __stdcall PlayerLaunch_AFTER(unsigned int iShipID, unsigned int iClientID)
+{
+	returncode = DEFAULT_RETURNCODE;
+
+	if (!iLoadedBurstFires)
+		return;
+
+	aClientData[iClientID].mapBurstFills.clear();
+	int i = 0;
+
+	IObjInspectImpl *obj = HkGetInspect(iClientID);
+	if (obj)
+	{
+		CShip* cship = (CShip*)HkGetEqObjFromObjRW((IObjRW*)obj);
+		CEquipManager* eqmanager = (CEquipManager*)((char*)cship + 0xE4);
+		CEquipTraverser tr(-1);
+		CEquip *equip = eqmanager->Traverse(tr);
+		while (equip)
+		{
+			if (CEGun::cast(equip))
+			{
+				CEGun* cgun = (CEGun*)equip;
+				map<unsigned int, struct AmmoGenerator>::iterator iter = mapBurstFire.find(cgun->GunArch()->iArchID);
+				if (iter != mapBurstFire.end())
+				{
+					ConPrint(L"BURSTFIRE: PlayerLaunch_AFTER - Iteration %d for client %u, removing 200 units of 0x%08X.\n", i, iClientID, iter->second.ammo);
+					//HkRemoveCargo((const wchar_t*)Players.GetActiveCharacterName(iClientID), iter->second.ammo, 2000);	// better safe than sorry
+					aClientData[iClientID].mapBurstFills[i].ammo = iter->second.ammo;
+					aClientData[iClientID].mapBurstFills[i].count = iter->second.count;
+					aClientData[iClientID].mapBurstFills[i].timestamp = 0;
+					i++;
+				}
+			}
+			equip = eqmanager->Traverse(tr);
+		}
+	}
+
+	list<CARGO_INFO> lstCargo;
+	int iRemainingHoldSize = 0;
+	if (HkEnumCargo((const wchar_t*)Players.GetActiveCharacterName(iClientID), lstCargo, iRemainingHoldSize) == HKE_OK)
+	{
+		foreach(lstCargo, CARGO_INFO, item)
+		{
+			for (map<unsigned int, struct AmmoGenerator>::iterator iter = mapBurstFire.begin(); iter != mapBurstFire.end(); ++iter)
+				if (item->iArchID == iter->second.ammo)
+					HkRemoveCargo((const wchar_t*)Players.GetActiveCharacterName(iClientID), item->iID, item->iCount);
+		}
+	}
+}
+
 void HkTimerCheckKick()
 {
 	returncode = DEFAULT_RETURNCODE;
 
 	uint curr_time = (uint)time(0);
-	static int messageno = 1;
+	/*static int messageno = 1;
 
 	wchar_t message[1024];
-	swprintf_s(message, L"Hi! I'm an FLHook plugin! This is message number %d.", messageno++);
+	swprintf_s(message, L"Hi! I'm an FLHook plugin! This is message number %d.", messageno++);*/
 	//WriteKillposterSocket(message);
 }
 
+void HkTimerNPCAndF1Check()
+{
+	returncode = DEFAULT_RETURNCODE;
+
+	if (!iLoadedBurstFires)
+		return;
+
+	mstime curr_mstime = GetTimeInMS();
+	//ConPrint(L"BURSTFIRE: Tick %llu.\n", curr_mstime);
+	for (int i = 0; i < 250; i++)
```

Comment:
same as earlier with the max client ID thing

Reply:
It's somewhere in there, I'll go dig it up